javascript/react/wagmi.sh/src/DeployLock.js

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react'
+import { ethers } from 'ethers'
+import { UnlockV11, PublicLockV11 } from "@unlock-protocol/contracts";
+
+import {
+  erc20ABI,
+  useAccount,
+  useSendTransaction,
+  useWaitForTransaction,
+  useContractRead,
+  usePrepareContractWrite
+} from 'wagmi'
+
+const lockInterface = new ethers.utils.Interface(PublicLockV11.abi)
+
+
+export function DeployLock() {
+  const { address: creator } = useAccount()
+
+  const [calldata, setCalldata] = React.useState('')
+  const [name, setName] = React.useState('My Membership')
+  const [price, setPrice] = React.useState(1)
+  const [duration, setDuration] = React.useState(30) // in days
+  const [supply, setSupply] = React.useState(10000)
+  const [currency, setCurrency] = React.useState('') // address of the ERC20. If 0x0, uses base currency
```

Comment:
wanna replace this with `react-hook-form` for improved readability or we wanna keep this very minimal?

Reply:
I wanted to limit the dependencies to keep things as "vanilla" as possible...